sklearnex/neighbors/knn_classification.py

```diff
@@ -15,4 +15,156 @@
 # limitations under the License.
 #===============================================================================
 
-from daal4py.sklearn.neighbors import KNeighborsClassifier
+from distutils.version import LooseVersion
+from sklearn import __version__ as sklearn_version
+
+from sklearn.neighbors._base import KNeighborsMixin as sklearn_KNeighborsMixin
+from sklearn.neighbors._base import RadiusNeighborsMixin as sklearn_RadiusNeighborsMixin
+from sklearn.neighbors._base import NeighborsBase as sklearn_NeighborsBase
+from sklearn.neighbors._ball_tree import BallTree
+from sklearn.neighbors._kd_tree import KDTree
+from sklearn.neighbors._base import _check_weights
+
+from sklearn.neighbors._classification import KNeighborsClassifier as \
+    sklearn_KNeighborsClassifier
+
+from sklearn.utils.validation import _deprecate_positional_args
+
+from onedal.datatypes import _check_array
+
+from onedal.neighbors import KNeighborsClassifier as onedal_KNeighborsClassifier
+from onedal.neighbors import KNeighborsMixin as onedal_KNeighborsMixin
+
+from .._device_offload import dispatch, wrap_output_data
+import numpy as np
+from scipy import sparse as sp
+
+
+class KNeighborsMixin(sklearn_KNeighborsMixin):
+    def kneighbors(self, X=None, n_neighbors=None, return_distance=True):
+        onedal_model = getattr(self, '_onedal_model', None)
+        if X is not None:
+            X = _check_array(
+                X, accept_sparse='csr', dtype=[
+                    np.float64, np.float32])
+
+        if onedal_model is not None and not sp.issparse(X):
+            result = onedal_KNeighborsMixin._kneighbors(self, X, n_neighbors, return_distance)
+

```

Comment:
What about passing a queue to onedal and overall device support?
Make it consistent with other functions, use `dispatch()` method